dpypelines/pipeline/dataset_ingress_v1.py

```diff
@@ -34,118 +35,121 @@ def dataset_ingress_v1(files_dir: str, pipeline_config: dict):
             de_notifier.failure()
-            raise Exception(
-                message.unexpected_error(
-                    f"Error while looking for required file {required_file}", err
-                )
-            ) from err
+            raise err
 
     # Extract the patterns for supplementary distributions from the pipeline configuration
     try:
         supplementary_distribution_patterns = (
             matching.get_supplementary_distribution_patterns(pipeline_config)
         )
+        logger.info("Successfully retrieved supplementary distribution patterns from pipeline config",
+         data={"supplementary_distribution_pattenrs": supplementary_distribution_patterns})
     except Exception as err:
+        files_in_directory = local_store.get_file_names()
+        logger.error("Failed to get supplementary distribution patterns", err, data={"pipeline_config": pipeline_config})
         de_notifier.failure()
-        raise Exception(
-            message.unexpected_error(
-                "Failed to get supplementary distribution patterns", err
-            )
-        ) from err
+        raise err
 
     # Check for the existence of each supplementary distribution
     for supplementary_distribution in supplementary_distribution_patterns:
         try:
             if not local_store.has_lone_file_matching(supplementary_distribution):
-                de_notifier.failure()
-                msg = message.expected_local_file_missing(
-                    f"Supplementary distribution {supplementary_distribution} not found",
-                    supplementary_distribution,
-                    "dataset_ingress_v1",
-                    local_store,
-                )
-                raise ValueError(msg)
+                raise FileNotFoundError(f"Could not find file matching pattern {supplementary_distribution}")
         except Exception as err:
+            logger.error(f"Error while looking for supplementary distribution {supplementary_distribution}", 
+            err, 
+            data={"supplementary_distribution": supplementary_distribution,
+            "supplementary_distribution_patterns": supplementary_distribution_patterns}
+            )
             de_notifier.failure()
-            raise Exception(
-                message.unexpected_error(
-                    f"Error while looking for supplementary distribution {supplementary_distribution}",
-                    err,
-                )
-            ) from err
+            raise err
 
     # Get the positional arguments (the inputs) from the pipeline_config
     # dict and run the specified sanity checker for it
     args = []
     for match, sanity_checker in pipeline_config["transform_inputs"].items():
         try:
             input_file_path: Path = local_store.save_lone_file_matching(match)
+            logger.info("Successfully saved file that matches pattern", data={"input_file_path": input_file_path, "match": match})
```

Comment:
can you include files_in_directory again please, so we saved X to Y from. a choice. of Z, that'd be everything you might want to know.